src/odoc.ml

```diff
@@ -267,17 +285,14 @@ let gen_rules sctx ~dir:_ rest =
     setup_css_rule sctx;
     setup_toplevel_index_rule sctx
   | lib :: _ ->
-    let libs = SC.libs sctx in
-    let (lib, scope) =
+    let lib, lib_db =
       match String.rsplit2 lib ~on:'@' with
-      | None -> (lib, Lib_db.external_scope libs)
-      | Some (lib, name) -> (lib, Lib_db.find_scope_by_name_exn libs ~name)
+      | None ->
+        (lib, SC.public_libs sctx)
+      | Some (lib, name) ->
+        (lib, Scope.libs
+                (SC.find_scope_by_name sctx (Scope_info.Name.of_string name)))
     in
-    let scope =
-      { With_required_by.
-        data = scope
-      ; required_by = [Alias (Path.of_string "doc")]
-      } in
-    let open Option.Infix in
-    Option.iter (Lib_db.Scope.find scope lib >>= Lib.src_dir)
-      ~f:(fun dir -> SC.load_dir sctx ~dir)
+    match Lib.DB.find lib_db lib with
+    | Error _ -> ()
```

Comment:
A bit off topic since that's the old behavior anyway, but why are we ignoring such errors? This cames seems impossible, so shouldn't it be assert false or a Sexp.code_error?

Reply:
Actually it's not impossible. We can reach this case with:

```sh
$ jbuilder build _build/default/_doc/blah/...
```

Currently, this will cause a `no rule found for ...` error. Maybe we could return something more precise